backend/api/Services/LocalizationService.cs

```diff
@@ -0,0 +1,230 @@
+    {
+        private readonly IInstallationService _installationService;
+        private readonly ILogger<LocalizationService> _logger;
+        private readonly IMissionRunService _missionRunService;
+        private readonly IRobotService _robotService;
+
+        public LocalizationService(ILogger<LocalizationService> logger, IRobotService robotService, IMissionRunService missionRunService, IInstallationService installationService)
+        {
+            _logger = logger;
+            _robotService = robotService;
+            _missionRunService = missionRunService;
+            _installationService = installationService;
+        }
+
+        public async Task EnsureRobotIsOnSameInstallationAsMission(Robot robot, MissionDefinition missionDefinition)
+        {
+            var missionInstallation = await _installationService.ReadByName(missionDefinition.InstallationCode);
+            var robotInstallation = await _installationService.ReadByName(robot.CurrentInstallation);
+
+            if (missionInstallation is null || robotInstallation is null)
+            {
+                string errorMessage = $"Could not find installation for installation code {missionDefinition.InstallationCode}";
+                _logger.LogError("{Message}", errorMessage);
+                throw new InstallationNotFoundException(errorMessage);
+            }
+
+            if (robotInstallation != missionInstallation)
+            {
+                string errorMessage = $"The robot {robot.Name} is on installation {robotInstallation.Name} which is not the same as the mission installation {missionInstallation.Name}";
+                _logger.LogError("{Message}", errorMessage);
+                throw new MissionException(errorMessage);
+            }
+        }
+
+        public async Task EnsureRobotIsCorrectlyLocalized(Robot robot, MissionDefinition missionDefinition)
+        {
+            if (missionDefinition.Area is null)
+            {
+                string errorMessage = $"There was no area associated with mission definition {missionDefinition.Id}";
+                _logger.LogError("{Message}", errorMessage);
+                throw new AreaNotFoundException(errorMessage);
+            }
+
+            if (!await RobotIsLocalized(robot.Id)) { await StartLocalizationMissionInArea(robot, missionDefinition.Area); }
+
+            if (!RobotIsOnSameDeckAsMission(robot.CurrentArea, missionDefinition.Area))
+            {
+                string errorMessage = $"A new mission run of mission definition {missionDefinition.Id} will not be started as the robot is not localized on the same deck as the mission";
+                _logger.LogError("{Message}", errorMessage);
+                throw new RobotLocalizationException(errorMessage);
+            }
+        }
+
+        public async Task<bool> RobotIsLocalized(string robotId)
+        {
+            var robot = await _robotService.ReadById(robotId);
+            if (robot != null)
```

Comment:
Any reason for having the return inside the if instead of the throwing of the exception?
Given that in most places we do the opposite this looks less readable to me